Dockerfile

```diff
@@ -1,31 +1,38 @@
-# Start by building the sta-rs library in a builder container.
-FROM rust:1.60 as rust-builder
+# Start by building the nitriding proxy daemon.
+FROM public.ecr.aws/docker/library/golang:1.20 as go-builder
+
+WORKDIR /src/
+COPY . .
+RUN make -C nitriding/cmd nitriding
+
+# Build the web server application itself.
+# Use the -alpine variant so it will run in a alpine-based container.
+FROM public.ecr.aws/docker/library/rust:1.68.2-alpine as rust-builder
```

Comment:
Is the pin on the minor version needed here? Normally we'd ask to use `1.68-alpine` here in order to get any minor fixes / patches, but maybe there's attestation requirements that mean we need to pin to the full version?

Reply:
Assume we start an enclave based on x.y-alpine. Shortly after, x.y.z-alpine comes out. A user is trying to verify our deployment based on the newly-release x.y.z-alpine and notices that it fails because the enclave still runs x.y-alpine.

We could work around that by updating the enclave whenever a new patch comes out.

An even better solution would be to get rid of alpine altogether and only have the statically-compiled nitriding and star-randsrv in the image. For now, we only need alpine because of the shell script that starts nitriding and star-randsrv — a very heavy dependency for a simple task. I filed https://github.com/brave/nitriding/issues/58 for this improvement.

Reply:
Hopefully tls-alpn means we don't need a trusted ca certificate package either.

Reply:
@fmarier Generally I agree that specifying a particular point release tag bitrots quickly and misses out on useful fixes. Because in the particular application reproducible builds are important, tracking the specific versions of our dependencies, as we do with `go.sum` and `Cargo.lock` is a worthwhile trade-off.

Hopefully we can use renovatebot or similar to make sure we bump regularly.